RGL1.py

```diff
@@ -0,0 +1,142 @@
+from typing import Union
+from Bio import SeqIO, SeqUtils
+from abc import ABC, abstractmethod
+
+
+def length_filter(record, lower_length_bound: int, upper_length_bound: int):
+    if lower_length_bound <= len(record.seq) <= upper_length_bound:
+        return record
+
+
+def gc_filter(record, lower_gc_bound: float, upper_gc_bound: float):
+    if lower_gc_bound <= SeqUtils.GC(record.seq) <= upper_gc_bound:
+        return record
+
+
+def quality_filter(record, quality_threshold: Union[int, float] = 0):
+    if (quality_threshold < sum(record.letter_annotations['phred_quality']) /
+            len(record.letter_annotations['phred_quality'])):
+        return record
+
+
+def filter_fastq(file_path: str, output_file: str = 'input_fasta_name_filter.fastq',
+                 lower_gc_bound: int = 0, upper_gc_bound: int = 30,
+                 lower_length_bound: int = 0, upper_length_bound: int = 2**32,
+                 quality_threshold: Union[int, float] = 0) -> None:
+    records = SeqIO.parse(file_path, 'fastq')
+    if output_file == 'input_fasta_name_filter.fastq':
+        output_filename = file_path.split('\\')[-1]
+        output_file = output_filename.replace('.fastq', '_filter.fastq')
+    with open(output_file, 'w') as output:
+        for record in records:
+            if length_filter(record, lower_length_bound, upper_length_bound) and \
+                    gc_filter(record, lower_gc_bound, upper_gc_bound) and \
+                    quality_filter(record, quality_threshold):
+                SeqIO.write(record, output, 'fastq')
+
+
+class BiologicalSequence(ABC):
+    @abstractmethod
+    def __len__(self):
+        pass
+
+    @abstractmethod
+    def __getitem__(self, item: int):
+        pass
+
+    @abstractmethod
+    def __str__(self):
+        pass
+
+    @abstractmethod
+    def check_alphabet(self):
+        pass
+
+
+class SequenceFunction(BiologicalSequence):
+    alphabet = ()
+
+    def __init__(self, seq: str) -> None:
+        self.seq = seq
+
+    def __len__(self) -> int:
+        self.length = len(self.seq)
+        return self.length
+
+    def __getitem__(self, item: int) -> str:
+        if 0 <= item < len(self.seq):
+            return self.seq[item]
+        else:
+            raise IndexError('Your index is incorrect')
+
+    def __str__(self) -> str:
+        return str(self.seq)
+
+    def check_alphabet(self) -> bool:
+        return set(self.seq).issubset(set(self.alphabet))
+
+
+class NucleicAcidSequence(SequenceFunction):
+    complement_dict = {}
+    def __init__(self, seq: str):
+        super().__init__(seq)
+
+    def complement(self) -> str:
+        complement_seq = self.seq.translate(str.maketrans(self.complement_dict))
+        return complement_seq
+
+    def gc_content(self) -> Union[int, float]:
+        gc_count = self.seq.count('C') + self.seq.count('G')
+        return gc_count/len(self.seq)*100
+

```

Comment:
Аннотация типа возвращает Union[int, float], но на практике это всегда будет float
```suggestion
def gc_content(self) -> float:
```